pkg/cmd/codespace/select.go

```diff
@@ -0,0 +1,72 @@
+package codespace
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+type selectOptions struct {
+	filePath string
+}
+
+func newSelectCmd(app *App) *cobra.Command {
+	opts := selectOptions{}
+
+	selectCmd := &cobra.Command{
+		Use:   "select",
+		Short: "Select a codespace",
+		Hidden: true,
+		Args:  noArgsConstraint,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return app.Select(cmd.Context(), "", opts)
+		},
+	}
+
+	selectCmd.Flags().StringVarP(&opts.filePath, "file", "f", "", "Output file path")
+	return selectCmd
+}
+
+// Hidden codespace `select` command allows to reuse existing codespace selection
+// dialog by external GH CLI extensions. By default output selected codespace name
+// into `stdout`. Pass `--file`(`-f`) flag along with a file path to output selected
+// codespace name into a file instead.
+// 
+// ## Examples
+//
+// With `stdout` output:
+//
+// ```shell
+// 		gh codespace select
+// ```
+//
+// With `into-a-file` output:
+//
+// ```shell
+// 		gh codespace select --file /tmp/selected_codespace.txt
+// ```
+func (a *App) Select(ctx context.Context, name string, opts selectOptions) (err error) {
+	codespace, err := getOrChooseCodespace(ctx, a.apiClient, name)
+	if err != nil {
+		return err
+	}
+
+	if opts.filePath != "" {
+		f, err := os.Create(opts.filePath)
+		if err != nil {
+			return fmt.Errorf("failed to create output file: %w", err)
+		}
+
+		f.WriteString(codespace.Name);
+		
+		defer safeClose(f, &err)
```

Comment:
Flip these two, and also `os.File.WriteString` returns an error that we should check. That's why we should defer the close before we attempt to do anything with it.

Reply:
I see what you meant by `safeClose` now 😊 Fixed.